added oatk tool

Deeptivarshney opened the pull request:
FOR CONTRIBUTOR:
* [ x] I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] License permits unrestricted use (educational + commercial)
* [x] This PR adds a new tool or tool collection
* [ ] This PR updates an existing tool or tool collection
* [ ] This PR does something else (explain below)

Closes [#6708](https://github.com/galaxyproject/tools-iuc/issues/6708)

Deeptivarshney committed:
added otak_tool

shiltemann committed:
add sections to test parameters

shiltemann committed:
indent

shiltemann committed:
not in biotools

shiltemann committed:
fix dependency versions

shiltemann committed:
add citation

shiltemann committed:
use variables for inputs, fix filetype

shiltemann committed:
use hmmpress on anontation files before oatk command

shiltemann committed:
allow some differences

shiltemann committed:
index files do not need to be included

shiltemann committed:
remove non working test data

shiltemann committed:
downsize test dataset, remove 1000trimmed dataset

shiltemann committed:
deal with optional inputs, set diff in test

shiltemann committed:
add missing parameter in command

shiltemann committed:
filter outputs, add tests

shiltemann committed:
add output for added test

shiltemann committed:
change value of booleans in test for planemo lint

shiltemann committed:
this parameter is not in the help anymore, so remove it

shiltemann commented on tools/oatk/oatk.xml:
```diff
@@ -0,0 +1,209 @@
+        <!-- Annotation Options (hmmannot) -->
+        <section name="Annotation" title="Annotation Options" expanded="False">
+            <param name="annotation_hmm_mito" type="data" format="hmm3" optional="true" label="Mitochondria HMM Profile Database"/>
+            <param name="annotation_hmm_plastid" type="data" format="hmm3" optional="true" label="Plastid HMM Profile Database"/>
```
@Deeptivarshney are these annotation files something that users should supply themselves, or should we have these files (and others?) as built-in options for them?

Deeptivarshney replied:
If we create a data manager for that, we would need to keep track of version updates if we go that direction. So maybe for the first release on Galaxy, we can just let users supply those files.

shiltemann committed:
temp dir shouldnt be a user option

shiltemann committed:
executable path shouldnt be user option

shiltemann committed:
lowercase tool name

shiltemann committed:
remote repo url should point to IUC folder

shiltemann committed:
leading spaces

shiltemann committed:
update test data to avoid empty test output files

shiltemann committed:
update test data to avoid empty test output files

shiltemann committed:
parameter under development, remove for now

shiltemann committed:
add tip about where to find annotation files

shiltemann committed:
remove debug line

SaimMomin12 commented:
> oh, file size limit is 1MB, I thought 5MB, will have to downsample more later, @Deeptivarshney if you happen to have some small input and annotation files that still produce non-empty outputs, let me know

@shiltemann Alternatively, you can also upload your test data to our [Zenodo Tool Developers Community.](https://zenodo.org/communities/galaxy_tool_dev_sig/records?q=&l=list&p=1&s=10&sort=newest)

Deeptivarshney committed:
Merge branch 'galaxyproject:main' into oatk_tool

Deeptivarshney committed:
added Zenodo link for testdata

Deeptivarshney committed:
Update tools/oatk/oatk.xml

Co-authored-by: Saim Momin <64724322+SaimMomin12@users.noreply.github.com>

shiltemann committed:
Update tools/oatk/oatk.xml

Co-authored-by: Saim Momin <64724322+SaimMomin12@users.noreply.github.com>

SaimMomin12 commented on tools/oatk/oatk.xml:
```diff
@@ -0,0 +1,207 @@
+            </section>
+            <section name="Pathfinder">
+                <param name="include_trna" value="True" />
+                <param name="include_rrna" value="True" />
```
```suggestion
                <param name="include_trna" value="true" />
                <param name="include_rrna" value="true" />
```

SaimMomin12 reviewed (commented):
Some minor comments inline

shiltemann committed:
Apply suggestions from code review

Co-authored-by: Saim Momin <64724322+SaimMomin12@users.noreply.github.com>

shiltemann committed:
Apply suggestion from @shiltemann

shiltemann committed:
indent code block

Deeptivarshney committed:
Merge branch 'galaxyproject:main' into oatk_tool

Deeptivarshney committed:
Update parameter labels in oatk.xml

scorreard reviewed (commented):
I think these should solve the test issues.
- One test file was not correct
- The test output where not pointing to the correct files

shiltemann committed:
Apply suggestions from code review

Co-authored-by: Solenne Correard <54953390+scorreard@users.noreply.github.com>

Deeptivarshney committed:
Update tools/oatk/oatk.xml

Co-authored-by: Solenne Correard <54953390+scorreard@users.noreply.github.com>

Deeptivarshney committed:
Remove test-data dir

scorreard commented on tools/oatk/oatk.xml:
```diff
@@ -0,0 +1,305 @@
+    --edge-c-tag ${edge_coverage_tag}
+    --kmer-c-tag ${kmer_coverage_tag}
+    --seq-c-tag ${seq_coverage_tag}
+    "${input_reads}"
```
```suggestion
    '${input_reads}'
```

scorreard commented on tools/oatk/oatk.xml:
```diff
@@ -0,0 +1,305 @@
+        <section name="Annotation" title="Annotation Options" expanded="false">
+            <param name="annotation_hmm_mito" type="data" format="hmm3" optional="true" label="Mitochondria HMM Profile Database" help="you must supply at least one HMM Profile Database (mitochondria or plastids). These can be obtained from OatkDB (see help below this tool for link)"/>
+            <param name="annotation_hmm_plastid" type="data" format="hmm3" optional="true" label="Plastid HMM Profile Database"/>
+            <param name="batch_size" type="integer" value="100000" min="1" label="Batch Size"/>
```
@Deeptivarshney do you know? You included more parameters that I can find on their GitHub, but you may have the experience from using the tool.

Deeptivarshney replied:
Thanks both, I’ll pick this up next week; I am a bit busy and need to finish couple of things this week.

Deeptivarshney replied:
@bgruening, This `batch_size` parameter checks how many sequences are processed per batch during the HMM annotation step. It mainly affects performance: larger batch sizes can improve speed but use more memory. Since it is a technical parameter and most users would not tune it, I guess... (or somehow we can restrict it...) Or do you think it’s better to keep it as a default, or can we move it to advanced options?

Deeptivarshney replied:
Hi @shiltemann & @SaimMomin12, Could you also suggest what would be the best way to deal with such parameter?

scorreard commented on tools/oatk/oatk.xml:
```diff
@@ -0,0 +1,305 @@
+            <param name="smer_size" type="integer" value="31" min="1" max="31" label="S-mer Size"/>
+            <param name="min_kmer_coverage" type="integer" value="30" min="1" label="Minimum K-mer Coverage"/>
+            <param name="min_arc_coverage" type="float" value="0.35" min="0" label="Minimum Arc Coverage"/>
+            <param name="max_data_usage" type="text" value="0" label="Maximum Data Usage (e.g., 0, 10K, 10M)"/>
```
@Deeptivarshney do you know? You included more parameters that I can find on their GitHub, but you may have the experience from using the tool.

Deeptivarshney replied:
> what does this param do? Should it be better an integer?
About `max_data_usage` parameter.. It supports suffixes like K/M/G (e.g. 10G), so I used type="text" to allow those values in Galaxy

Deeptivarshney committed:
Update tools/oatk/oatk.xml

Co-authored-by: Solenne Correard <54953390+scorreard@users.noreply.github.com>

Deeptivarshney committed:
help section for Syncasm options

Deeptivarshney committed:
help section for Annotation options

Deeptivarshney committed:
help section for pathfinder options

bgruening commented on tools/oatk/oatk.xml:
```diff
@@ -0,0 +1,305 @@
+      -p plastid.fam
+    #end if
+    ${use_graph_mode}
+    -k ${kmer_size}
```
you need to reference here the section as `${Syncasm.kmer_size}`, ideally with `${syncasm.kmer_size}`

bgruening reviewed (commented):
* The `oatk` command uses `-o PREFIX` to set the output file prefix (default: `./oatk.asm`). The command block never passes it explicitly, making the wrapper fragile — any change in the tool's default would silently break all output collection. Something like that maybe?

```xml
-o oatk.asm
```
* The input parameter `max_graph_paths` is defined (mapped to the `-N` flag) but is completely absent from the command block. The tool never receives this value.

* `type="txt"` is not a valid Galaxy param type you mean `type="text"` I guess

* Several `type="text"` parameters are interpolated **unquoted** into the command line:

```xml
-D ${max_data_usage}
-g ${min_core_gene_gain}
--edge-c-tag ${edge_coverage_tag}
--kmer-c-tag ${kmer_coverage_tag}
--seq-c-tag ${seq_coverage_tag}
```

A user entering `; rm -rf /` for `max_data_usage` would execute arbitrary shell commands.

Do something like:
```xml
<param argument="-D" name="max_data_usage" type="text" value="0" label="Maximum Data Usage">
    <validator type="regex" message="Enter a number optionally followed by K, M, or G">^[0-9]+[KMG]?$</validator>
</param>
```

* OATK produces GFA 1.x format files (the test assertions confirm `VN:Z:1.0` in the header). `gfa2` is the GFA 2.0 format, a different. Please double check.

* The input declares `format="fastq,fasta,fastq.gz"`. The bare `fastq` type is a legacy type that does not participate in Galaxy format detection. 

Maybe:
```xml
format="fastqsanger,fastqsanger.gz,fasta,fasta.gz"
```

* IUC convention requires `argument="-k"` (or the long-form flag) on each `<param>` so the Galaxy UI can show the underlying CLI flag. Almost no parameters in this wrapper carry `argument=`.

* IUC tools should use `label="${tool.name} on ${on_string}: description"` so users can track provenance across multiple runs. Current labels are freestanding strings like `"The GFA file for the final genome assembly"`.

* Output names mix `snake_case` (`final_gfa_file`, `mt_gene`) with `PascalCase` (`MT_subgraph`, `PT_structure_solved`). All output names should follow lowercase `snake_case` per IUC standard.

* The help text states "you must supply at least one HMM Profile Database", but both HMM inputs are `optional="true"`

* Parameters defined inside `<section>` elements (e.g., `Syncasm`, `Pathfinder`, `Annotation`) are referenced flat in the command block (e.g., `${kmer_size}` instead of `${Syncasm.kmer_size}`). This works in Galaxy when names are unique, but is inconsistent style. The preferred convention is to use section-prefixed references.

`categories: [Genome annotation]` is not appropriate for an assembler. The comparable `hifiasm` tool uses `Assembly`.

* Test 3 sets `expect_num_outputs="11"` but only asserts on 10 outputs — `final_gfa_file` has no content assertion. It should be included for consistency with tests 1 and 2.

* There are a few whitespace incosistencies, please run the VSC extension and reformat everything

* `minicircle_mode` is commented out
* `"...see help below this tood for link"` — "tood" should be "tool".
- Citation: The DOI `10.5281/zenodo.7631376` points to a Zenodo software record. If the OATK paper has been published, its DOI is preferred.

scorreard committed:
Implemented part of the feedback from Bjorn

Updated input and output parameters for the oatk tool, including changes to file formats and labels. Adjusted Syncasm and Pathfinder options for improved clarity and consistency.

scorreard committed:
Add 'Assembly' category to oatk tool

scorreard committed:
Update oatk.xml

scorreard committed:
Update citation DOI in oatk.xml

Commit:
Updates of the wrapper

Deeptivarshney committed:
Merge pull request #1 from scorreard/oatk

OATK updates to reply to Bjorn feedbcak

SaimMomin12 commented on tools/oatk/oatk.xml:
```diff
@@ -0,0 +1,308 @@
+    <macros>
+        <token name="@TOOL_VERSION@">1.0</token>
+        <token name="@VERSION_SUFFIX@">0</token>
+        <token name="@PROFILE@">24.2</token>
```
```suggestion
        <token name="@PROFILE@">25.0</token>
```

SaimMomin12 commented on tools/oatk/oatk.xml:
```diff
@@ -0,0 +1,308 @@
+    #end if
+
+    oatk
+    -t \${GALAXY_SLOTS:-2}
```
```suggestion
    -t "\${GALAXY_SLOTS:-2}"
```

SaimMomin12 reviewed (commented):
Can we format this XML using `planemo format oatk.xml`?

Deeptivarshney committed:
Update tools/oatk/oatk.xml

Co-authored-by: Saim Momin <64724322+SaimMomin12@users.noreply.github.com>

Deeptivarshney committed:
Update tools/oatk/oatk.xml

Co-authored-by: Saim Momin <64724322+SaimMomin12@users.noreply.github.com>

scorreard committed:
ran planemo format oatk.xml

Deeptivarshney committed:
Merge pull request #2 from scorreard/patch-1

ran planemo format oatk.xml

mvdbeek committed:
Fix oatk HMM profile datatype and container resolution

The `.fam` profile databases from OatkDB are HMMER3 files (they start
with `HMMER3/f`), so they sniff as the `hmm3` datatype. `fam` is not a
registered Galaxy datatype, so drop it from the accepted formats and
set explicit ftypes on the test inputs fetched from Zenodo.

Also drop the redundant hmmer requirement: bioconda's oatk package
already depends on `hmmer >=3.4`, and listing it separately made Galaxy
look for a mulled image that was never built, leaving the job without a
container (exit code 127). The single oatk biocontainer provides
hmmpress and hmmsearch.

Deeptivarshney committed:
Merge pull request #3 from mvdbeek/oatk_datatype_fix

Fix oatk HMM profile datatype and container resolution

Deeptivarshney committed:
added conditional param for hmm db

SaimMomin12 reviewed (commented):
LGTM! Some comments inline, feel free to accept them and i think we are then ready for a merge!

Deeptivarshney committed:
Update tools/oatk/oatk.xml

Co-authored-by: Saim Momin <64724322+SaimMomin12@users.noreply.github.com>

Deeptivarshney committed:
Update tools/oatk/oatk.xml

Co-authored-by: Saim Momin <64724322+SaimMomin12@users.noreply.github.com>

Deeptivarshney committed:
Update tools/oatk/oatk.xml

Co-authored-by: Saim Momin <64724322+SaimMomin12@users.noreply.github.com>

Deeptivarshney committed:
Update tools/oatk/oatk.xml

Co-authored-by: Saim Momin <64724322+SaimMomin12@users.noreply.github.com>

Deeptivarshney committed:
Update tools/oatk/oatk.xml

Co-authored-by: Saim Momin <64724322+SaimMomin12@users.noreply.github.com>

Deeptivarshney committed:
Update tools/oatk/oatk.xml

Co-authored-by: Saim Momin <64724322+SaimMomin12@users.noreply.github.com>

Deeptivarshney committed:
Update tools/oatk/oatk.xml

Co-authored-by: Saim Momin <64724322+SaimMomin12@users.noreply.github.com>

Deeptivarshney committed:
Update tools/oatk/oatk.xml

Co-authored-by: Saim Momin <64724322+SaimMomin12@users.noreply.github.com>

Deeptivarshney committed:
updated help section

Deeptivarshney committed:
updated test1

Deeptivarshney committed:
updated test2

Deeptivarshney committed:
updated test3

SaimMomin12 reviewed (approved):
LGTM! I think we can now merge. Thanks everyone involved here, and apologies for any inconvenience caused.
